src/boundarySolver/physics/logLawBoundary.cpp

```diff
@@ -0,0 +1,61 @@
+#include "logLawBoundary.hpp"
+#include <utilities/mathUtilities.hpp>
+#include "finiteVolume/compressibleFlowFields.hpp"
+
+void ablate::boundarySolver::physics::LogLawBoundary::Setup(ablate::boundarySolver::BoundarySolver &bSolver) {
+    bSolver.RegisterFunction(UpdateBoundaryVel, this, {finiteVolume::CompressibleFlowFields::EULER_FIELD}, {finiteVolume::CompressibleFlowFields::VELOCITY_FIELD});
+}
+
+PetscErrorCode ablate::boundarySolver::physics::LogLawBoundary::UpdateBoundaryVel(PetscInt dim, const ablate::boundarySolver::BoundarySolver::BoundaryFVFaceGeom *fg,
+                                                                                  const PetscFVCellGeom *boundaryCell, const PetscInt *uOff, PetscScalar *boundaryValues,
+                                                                                  const PetscScalar *stencilValues, const PetscInt *aOff, PetscScalar *auxValues, const PetscScalar *stencilAuxValues,
+                                                                                  void *ctx) {
+    PetscFunctionBeginUser;
+    const PetscInt EULER_FIELD = 0;
+    const PetscInt VEL = 0;
+    PetscReal logLawVel[dim];
+    PetscArrayzero(logLawVel, dim);
+    PetscReal stencilVel[dim];
+    PetscReal stencilNormalVelocity[dim];
+    PetscReal logLaw;
+    // get the stencil velocity in normal coord.
+    PetscReal stencilDensity = stencilValues[uOff[EULER_FIELD] + finiteVolume::CompressibleFlowFields::RHO];
+    for (PetscInt d = 0; d < dim; d++) {
+        stencilVel[d] = stencilValues[uOff[EULER_FIELD] + finiteVolume::CompressibleFlowFields::RHOU + d] / stencilDensity;
+        PetscReal transformationMatrix[3][3];
+        utilities::MathUtilities::ComputeTransformationMatrix(dim, fg->normal, transformationMatrix);
+        ablate::utilities::MathUtilities::Multiply(dim, transformationMatrix, stencilVel, stencilNormalVelocity);
+        // calculate the new boundary velocity in normal coord.
+        if (d == 1) {
```

Comment:
If we this outside of the dimension loop, we could get rid of the if statements.